### DISCUSSION TITLE
Reducing Transmission's screen real estate usage and some other UI/UX enhancements

### DISCUSSION THREAD

**Discussion by josephj11** [transmission/transmission#6440] [Ideas]
I have a number of conceptually separate things in this post. If there's any real interest in any of them, I'd be happy to separate them out into separate posts/threads/discussions.

For reference, I am using Transmission 4.0.5 on Kubuntu Linux 22.04 (KDE).

I have been happily using Transmission for a lot more than a decade. However, it wastes a lot of precious screen real estate.

I often like to micromanage my torrents, so I have the main Transmission window open along with several torrent properties windows. These fill up the screen very quickly and do weird things as far as which ones appear in front of the others when they overlap. The latter got worse with Transmission 3.x and newer (or maybe, it's related to upgrading from Kubuntu 18.04 to 22.04).

What I would like would be:

- The ability to greatly reduce Properties windows vertical dimension. When there are few peers, it's mostly empty space.
- Along with the above, a vertical scrollbar could be added to accommodate more peers
- A horizontal scrollbar would also be nice although not as important
- Along with the above, the ability to reorder columns would be nice, so I see the columns I care about without scrolling horizontally
- Allow the main Transmission window to be smaller horizontally with a horizontal scrollbar when needed. It already supports this in the vertical dimension, so it might be easy to add this
- It would be great to add the Torrent menu from the Main window to the Properties window so I could do things like ask for more peers directly without having to find the torrent in the main window...
- I often sort peers on the Flags field when seeding. This lets me see individual peers without them jumping around so much in the list. This would be much better if the individual flags were in a different order so they would sort better. I haven't worked out all the details of what I want for this, but would be happy to do so if there's any possibility of it getting implemented. I'd start with moving T  and E to the end and somewhat separating UuDd and K? into separate columns. I want to end up being able to list leechers before seeders and peers that are active ahead of those that are inactive. I'm not sure how many columns this would need, but I'd like to keep it as close as possible to what it is now
- It would be great if Transmission could remember which property windows I have open and their positions and be able to restore them when it is restarted

**Comment by Pentaphon** [comment]
You might be interested in https://github.com/transmission/transmission/pull/5319

**Reply by josephj11** [reply]
That should help horizontally, but it doesn't address the property windows (at all?) and the huge amount of wasted vertical space when there are very few peers and most of the window is empty and can't be made smaller.

Also, even when there are a lot of peers, most of them are just sitting there doing nothing and the few sorted to the top are all I need to see most of the time.

**Reply by Pentaphon** [reply]
You can run a 3rd party UI like https://github.com/xavery/transgui or https://github.com/openscopeproject/TrguiNG which are very traditional looking. I think the devs are pretty much set on keeping the existing UI as it currently is but with modern tweaks like new buttons, dark mode, etc. 

I personally like to just install the daemon only and run TrguiNG as a WebUI. Makes everything very lightweight.

**Reply by josephj11** [reply]
Thanks. I will take a look at them. I haven't worked with those technologies before. I did program in Pascal when it was very new, but haven't touched it in decades.

The most important change I want is just to reduce the minimum vertical window size for the properties windows. Theoretically, that should have very little effect  on how everything else looks and works.
